src/main/java/com/labsynch/labseer/service/SaltServiceImpl.java

```diff
@@ -99,4 +169,369 @@ private void saveSalt(CmpdRegMolecule mol) throws IOException, CmpdRegMolFormatE
+		Collection<IsoSalt> isoSalts =  isoSaltQuery.getResultList(); 
+		Set<IsoSalt> isoSaltsSet = new HashSet<IsoSalt>(isoSalts);
+
+		Set<SaltForm> saltFormSet = new HashSet<SaltForm>();
+		// Getter of IsoSalt to SaltForm 
+
+		for (IsoSalt isoSalt : isoSaltsSet)
+		{
+			SaltForm tempSaltForm = isoSalt.getSaltForm();
+			saltFormSet.add(tempSaltForm);
+		}
+
+		// Getter of Lots from IsoSalt 
+
+		Set<Lot> lotSet = new HashSet<Lot>(); 
+		for (SaltForm saltForm : saltFormSet)
+		{
+			Set<Lot> tempLotSet = saltForm.getLots();
+			lotSet.addAll(tempLotSet);
+		}
+
+		int dependencyLotSize = lotSet.size();
+		if (dependencyLotSize > 0)
+		{
+			ErrorMessage error = new ErrorMessage();
+			error.setLevel("warning");
+			error.setMessage("This salt is referenced by " + String.valueOf(dependencyLotSize) + " lots.");
+			warnings.add(error);
+
+			// Check for linked containers
+			Set<String> batchCodeSet = new HashSet<String>();
+			for (Lot lot : lotSet)
+			{
+				batchCodeSet.add(lot.getCorpName());
+			}
+			CmpdRegBatchCodeDTO batchDTO = lotService.checkForDependentData(batchCodeSet);
+
+			if(batchDTO.getBatchCodes() != null && batchDTO.getBatchCodes().size() > 0)
+			{
+				// Adds Associated Batch Codes to Warning Report 
+				error = new ErrorMessage();
+				error.setLevel("warning");
+				error.setMessage("Associated Batch Codes: " + batchDTO.getBatchCodes().toString());
+				warnings.add(error);
+			}
+
+			if(batchDTO.getLinkedExperiments() != null && batchDTO.getLinkedExperiments().size() > 0)
+			{
+				// Adds Linked Experiments to Warning Report 
+				error = new ErrorMessage();
+				error.setLevel("warning");
+				error.setMessage("Associated Experiments: " + batchDTO.getLinkedExperiments().toString());
+				warnings.add(error);
+			}
+			
+			if( batchDTO.getLinkedContainers() != null  && batchDTO.getLinkedContainers().size() > 0)
+			{
+				error = new ErrorMessage();
+				error.setLevel("warning");
+				error.setMessage("Associated Containers: " + batchDTO.getLinkedContainers().toString());
+				warnings.add(error);
+			}
+
+			// Adds Summary As Warning Message
+			error = new ErrorMessage();
+			error.setLevel("warning");
+			error.setMessage("Dependency Report Summary: " + batchDTO.getSummary());
+			warnings.add(error);
+
+			// Check CReg Config to See If Salt Abbrev in Lab Corp Name
+			if(!newSalt.getAbbrev().equals(oldSalt.getAbbrev())){
+				if (!propertiesUtilService.getCorpBatchFormat().equalsIgnoreCase("cas_style_format"))
+				{
+					error = new ErrorMessage();
+					error.setLevel("warning");
+					error.setMessage("Lot corp names references salt abbreviations. Any associated lot corp names will be updated.");
+					warnings.add(error);
+				}
+
+			}
+		}
+
+
+		return warnings;
+	}
+
+	public void updateDependencies(Salt salt)
+	{
+		
+
+		// Get IsoSalts From Salt 
+		TypedQuery<IsoSalt> isoSaltQuery = IsoSalt.findIsoSaltsBySalt(salt);
+		Collection<IsoSalt> isoSalts =  isoSaltQuery.getResultList(); 
```

Comment:
hypernit: you could do this in a one-liner:
```
Collection<IsoSalt> isoSalts =  IsoSalt.findIsoSaltsBySalt(salt).getResultList();
```
This is typically what we do, and I find it a little easier to follow.